pyincore/analyses/core_cge_ml/corecgeml.py

```diff
@@ -0,0 +1,266 @@
+            This is a dictionary of 2D arrays with shape [n, (k_i, l_i)]. Each entry in the dictionary corresponds to a factor and each factor has k_i number of models.
+            It is assumed that the intercept term is included in the model coefficients and is at the 0th column.
+        base_cap_factors : List[np.ndarray]
+            This is a list of 1D array with each entry corresponding to a factor representing its base capital by sectors.
+            This is the base capital for each sector for a given factor.
+            The list would look like [(m_1, 1), (m_2, 1), ..., (m_n, 1)].
+        """
+
+        # check if the shape of base_cap and capital_shocks match
+        if base_cap.shape != capital_shocks.shape:
+            raise ValueError(
+                "The shape of base_cap and capital_shocks do not match. Base Cap shape {}, Capital Shocks shape {}".format(
+                    base_cap.shape, capital_shocks.shape
+                )
+            )
+
+        # Convert capital_shocks to capital percent loss and then to capital loss in dollar amount
+        capital_loss = (1 - capital_shocks) * base_cap
+
+        # add a bias term to the capital loss with value 1 resulting in a shape of (1, 1+K)
+        capital_loss = np.hstack((np.ones((1, 1)), capital_loss))
+
+        assert len(model_coeffs) == len(
+            base_cap_factors
+        ), "The length of model_coeffs and base_cap_factors do not match. required length {}, observed length {}".format(
+            len(model_coeffs), len(base_cap_factors)
+        )
+
+        predictions = {}
+        for factor_base_cap_before, (factor, factor_model_coeff) in zip(
+            base_cap_factors, model_coeffs.items()
+        ):
+            # multiply the capital loss with the model coefficients (k_i x 1) = (1, 1+K) . (k_i, l_i)
+            factor_capital_loss: np.ndarray = capital_loss.dot(factor_model_coeff.T).T
+
+            assert (
+                factor_capital_loss.shape == factor_base_cap_before.shape
+            ), "Number of sectors in models and base_cap_factors do not match. required shape {}, observed shape {}".format(
+                factor_capital_loss.shape, factor_base_cap_before.shape
+            )
+            # add the predicted change in capital stock to the base_cap_factors
+            factor_base_cap_after: np.ndarray = (
+                factor_base_cap_before + factor_capital_loss
+            )
+
+            predictions[factor] = {
+                "before": np.squeeze(factor_base_cap_before).tolist(),
+                "after": np.squeeze(factor_base_cap_after).tolist(),
+            }
+
+        constructed_outputs = self.construct_output(predictions)
+
+        if constructed_outputs.get("ds", None) is not None:
+            self.set_result_csv_data(
+                "domestic-supply",
+                pd.DataFrame(constructed_outputs["ds"]),
+                name=self.get_parameter("domestic_supply_fname")
```

Comment:
Is it where @navarroc suggested you could fix a portion of the fname for datawolf to pick it up? Maybe try
```
name = self.get_parameter("domestic_supply_fname") + "-domestic-supply"
```

Reply:
Ah nvm i saw you address this in the other SLC PR. Thanks!